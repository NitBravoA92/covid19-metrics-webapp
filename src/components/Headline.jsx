import PropTypes from 'prop-types';
import '../assets/css/headline.css';

const Headline = ({ srcImg, name, information }) => (
  <section className="headline">
    <div className="region image">
      <img src={srcImg} alt={name} className="opacity-5" />
    </div>
    <div className="region info">
      <h1 className="name">{ name }</h1>
      {
        information.map(
          ({ stats, text }) => (
            <p className="stats-counter" key={text}>
              { stats }
              {' '}
              { text }
            </p>
          ),
        )
      }
    </div>
  </section>
);

Headline.defaultProps = {
  srcImg: '',
  name: '',
  information: [],
};

Headline.propTypes = {
  srcImg: PropTypes.string,
  name: PropTypes.string,
  information: PropTypes.arrayOf(PropTypes.shape({
    stats: PropTypes.string,
    text: PropTypes.string,
  })),
};

export default Headline;
