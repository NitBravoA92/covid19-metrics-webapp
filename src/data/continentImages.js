import Africa from '../assets/images/Africa.png';
import Asia from '../assets/images/Asia.png';
import AustraliaOceania from '../assets/images/AustraliaOceania.png';
import Europe from '../assets/images/Europe.png';
import NorthAmerica from '../assets/images/NorthAmerica.png';
import SouthAmerica from '../assets/images/SouthAmerica.webp';
import NewContinentFake from '../assets/images/NewContinentFake.png';

const continentImages = {
  Africa,
  Asia,
  'Australia-Oceania': AustraliaOceania,
  Europe,
  'North America': NorthAmerica,
  'South America': SouthAmerica,
  'New Continent Fake': NewContinentFake,
};

export default continentImages;
