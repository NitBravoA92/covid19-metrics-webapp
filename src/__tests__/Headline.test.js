import "@testing-library/jest-dom/extend-expect";
import { render, screen } from "@testing-library/react";
import Headline from "../components/Headline";
import continentImages from "../data/continentImages";

describe("The Headline component", () => {
  test("should render correctly into the DOM", () => {
    const { headlineComponent } = render(
      <Headline
        srcImg={continentImages.Asia}
        name="Asia"
        information={[{ stats: "224,317", text: "cases" }]}
      />
    );
    expect(headlineComponent).toMatchSnapshot();
  });

  test("should render the text: Europe with an empty array in the information prop", () => {
    const props = {
      srcImg: "no-image",
      name: "Europe",
      information: [],
    };
    render(<Headline {...props} />);
    const headline = screen.getByText("Europe");
    expect(headline).toBeInTheDocument();
  });

  test("should render the text: South America with 3 elements in the information prop array", () => {
      render(
      <Headline
        srcImg={continentImages['South America']}
        name="South America"
        information={[
          { stats: "68,905,921", text: "total cases" },
          { stats: "1,359,165", text: "deaths" },
          { stats: "66,504,727", text: "recovered" },
        ]}
      />
    );
    const headline = screen.getByText("South America");
    expect(headline).toBeInTheDocument();
  });
});
