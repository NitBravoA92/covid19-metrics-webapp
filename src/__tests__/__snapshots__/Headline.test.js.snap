// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Headline component should render correctly into the DOM 1`] = `undefined`;
